### PR TITLE
angular: Migrate from TSLint to ESLint

### PR DESCRIPTION
**Pull request by vishal423**
- Migrate angular `TSLint` linting rules to their equivalent `ESLint` rules. Related to  #10105 
- Fix linting violations related to unused imports, unsafe methods. 
- To execute `codelyzer` rules that rely on the `TSLint`, fallback with `eslint-plugin-tslint`.

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by vishal423** [comment]
@jhipster/developers, this PR is ready for review. If current migration looks good, then, I will follow a similar approach to migrate React code over the weekend.

**Comment by vishal423** [comment]
@mraible, @deepu105, could you please review this as well?

File: generators/client/files-angular.js [hunk 1]
```diff
@@ -37,6 +37,8 @@ const files = {
                 'tsconfig.json',
                 'tsconfig-aot.json',
                 'tslint.json',
```
Comment (mraible) [inline generators/client/files-angular.js]: Why are you keeping `tslint.json`?

File: generators/client/templates/angular/package.json.ejs [hunk 2]
```diff
@@ -172,7 +176,7 @@ Remove it once the issue is closed
   <%_ } _%>
   "scripts": {
     "prettier:format": "prettier --write \"{,src/**/}*.{md,json,ts,css,scss,yml}\"",
-    "lint": "tslint --project tsconfig.json -e 'node_modules/**'",
+    "lint": "eslint . --ext .js,.ts",
```
Comment (mraible) [inline generators/client/templates/angular/package.json.ejs]: Do you need to exclude node_modules like previously?

Comment (vishal423) [reply generators/client/templates/angular/package.json.ejs]: yes, that's excluded via an entry in the `.eslintignore`

File: generators/client/files-angular.js (hunk 1 above)
Comment (vishal423) [reply generators/client/files-angular.js]: `codelyzer` rules still rely on the `tslint`. There is an open ticket to migrate to `eslint` https://github.com/mgechev/codelyzer/issues/763.

**Pull request merged**